test/integration/update_int_test.go

```diff
@@ -103,6 +104,11 @@ func (suite *UpdateIntegrationTestSuite) TestUpdateAvailable() {
 	ts := e2e.New(suite.T(), false)
 	defer ts.Close()
 
+	cfg, err := config.NewCustom(ts.Dirs.Config, singlethread.New(), true)
+	suite.Require().NoError(err)
+	defer cfg.Close()
+	cfg.Set(constants.AutoUpdateConfigKey, "false")
```

Comment:
Since autoupdate always happens, we have a config key to avoid going through with it. When that config key is enabled, we just output a notice that there's an update available. So this block effectively tests both that there's an update available, and that the config key prevents the update from happening.